File: leetcode/203. Remove Linked List Elements/203._Remove_Linked_List_Elements_1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode removeElements(ListNode head, int val) {
        if(head == null)
            return null;
        ListNode dummy = new ListNode(-1); //dummy node
        dummy.next = head;
        ListNode current = dummy;
        while(current != null && current.next != null) {
            if(current.next.val == val) {
                current.next = current.next.next;
            }
            else {
                current = current.next;
            }
        }
        return dummy.next;
    }
}

File: leetcode/203. Remove Linked List Elements/203._Remove_Linked_List_Elements_1.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        dummy = ListNode(-1)
        dummy.next = head
        
        last = dummy
        current = dummy.next
        
        while current:
            if current.val == val:
                last.next = current.next
            else:
                last = current
            current = current.next
        return dummy.next